### PR TITLE
Use SLF4J loggers and add support for named hierarchies

### PR DESCRIPTION
**Pull request by arielvalentin**
Our attempt to address issues https://github.com/colinsurprenant/redstorm/issues/110 https://github.com/colinsurprenant/redstorm/issues/111

### CODE REVIEW

**Comment by colinsurprenant** [comment]
LGTM. reviewing this I noticed I created the `Loggable` module to try and DRY this up a bit but for some reason, this was never completed and `Loggable` is unused.

If you feel like completing this that would be great, otherwise I'll merge this and we can create an issue to not forget to complete the `Loggable` thing.

**Comment by arielvalentin** [comment]
I will give it a shot

**Comment by arielvalentin** [comment]
I replaced the usage with Loggable module. I didn't think to write additional tests for it. Is that something you would like?

File: lib/red_storm/dsl/topology.rb [hunk 1]
```diff
@@ -102,7 +102,7 @@ def new_instance
       end
 
       def self.log
-        @log ||= Java::OrgApacheLog4j::Logger.getLogger(self.name)
+        @log ||= Java::OrgSlf4j::LoggerFactory.get_logger(self.name.gsub(/::/,'.'))
```
Comment (colinsurprenant) [inline lib/red_storm/dsl/topology.rb]: why not use Loggable here too?

**Comment by colinsurprenant** [comment]
good stuff, thanks. just curious per my inline comment why not use `Loggable` in the `Topology` class too?

**Comment by arielvalentin** [comment]
Oops!! I'll fix that. 

Thanks,

## Ariel

**Pull request closed without merging**